shared/src/iosTest/kotlin/co/touchlab/kmp/BaseModelTest.kt

```diff
@@ -0,0 +1,50 @@
+package co.touchlab.kmp
+
+import co.touchlab.kampstarter.BaseTest
+import co.touchlab.kampstarter.ktor.network
+import co.touchlab.kampstarter.models.BaseModel
+import co.touchlab.stately.freeze
+import co.touchlab.stately.isFrozen
+import kotlinx.coroutines.*
+import kotlin.coroutines.CoroutineContext
+import kotlin.test.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class BaseModelTest : BaseTest() {
+    @Test
+    fun testScopeIsolation() = runTest {
+        lateinit var networkContext: CoroutineContext
+        lateinit var controlContext: CoroutineContext
+        val model = BaseModel()
+
+        //Control: the child context should get frozen when we freeze the parent
+        model.scope.launch {
+            withContext(Dispatchers.Main) {
+                controlContext = kotlin.coroutines.coroutineContext
+                delay(1000)
+            }
+        }
+
+        //Test: child context shouldn't be frozen
```

Comment:
I'm going back and forth on whether I think it makes more sense to do the test and control as separate `@Test` functions. Having them together makes the parallel a bit more clear but it also turns this into a pretty long method.